statement.go

```diff
@@ -186,6 +204,28 @@ func (s *Stmt) bindTime(val driver.NamedValue, t Type, n int) (mapping.State, er
 	return state, nil
 }
 
+func toAnySlice(v any) ([]any, error) {
```

Comment:
Maybe we can unify (part of) this function with the `func extractSlice[S any](vec *vector, val S) ([]any, error)` function in `vector_setters.go`? And move it to that file, or to `value.go`.